### PR TITLE
IA-2120: Added cloudwatch event

### PR DESCRIPTION
**Pull request by yaythecloud**
Should I make the apex redirect cloudwatch event off by default so it doesn't change the original functionality of the module?

### CODE REVIEW

File: functions/apex-restart.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    response = client.update_service(
+        cluster=cluster,
+        service=service,
+        desiredCount=2,
+        taskDefinition=service,
+        networkConfiguration={
+            'awsvpcConfiguration': {
+                'subnets': [
+                    subnet_one,
+                    subne_two
```
Comment (bushong1) [inline functions/apex-restart.py:23]: typo

File: functions/apex-restart.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    subnet_one=os.environ.get('SUBNET_ONE')
+    subnet_two=os.environ.get('SUBNET_TWO')
+
+    print("replacing apex service...")
+    response = client.update_service(
+        cluster=cluster,
+        service=service,
+        desiredCount=2,
+        taskDefinition=service,
+        networkConfiguration={
```
Comment (bushong1) [inline functions/apex-restart.py:19]: Ideally, we'd just have a `forceNewDeployment=True`

Is it possible to _just_ pass `forceNewDeployment=True`, or if not, how about you make an initial describe call to fetch the existing config based on the cluster/service name, then use that to populate the `update_service` and avoid that extra config.

File: cloudwatch-event.tf [hunk 3]
```diff
@@ -0,0 +1,115 @@
+data "archive_file" "apex_restart" {
+  type        = "zip"
+  source_dir  = "../../functions/"
+  output_path = "../../functions/apex_restart.zip"
+}
+
+resource "aws_lambda_function" "apex_restart" {
```
Comment (bushong1) [inline cloudwatch-event.tf:7]: All these resources need to be conditional based on a var.  Maybe consider putting this in an internal module so all the resources can be `count = var.weekly_restart_enabled ? 1 : 0`'d with a single line (as opposed to having to jam that on every resource.  As long as you can keep the vars passed to the internal module to a minimum, i think that makes sense.  If you'd need to pass a dozen vars, maybe it's not worth it?

File: functions/apex-restart.py (hunk 2 above)
Comment (yaythecloud) [reply functions/apex-restart.py]: Removed the non required attributes

File: apex-restart/cloudwatch-event.tf [hunk 4]
```diff
@@ -0,0 +1,112 @@
+data "archive_file" "apex_restart" {
+  type        = "zip"
+  source_dir  = "../../apex-restart/functions/"
```
Comment (bushong1) [inline apex-restart/cloudwatch-event.tf:3]: `${path.module}/functions/`

**Review by bushong1 — APPROVED** [review]
Add readme, changelog

**Review by bushong1 — APPROVED** [review]

**Pull request merged**